algorithms.js

```diff
@@ -1,14 +1,14 @@
 // 1. LINEAR SEARCH (returns the index of an array item)
 
-const linearSearch = (array, item) => {
+const linearSearch = (array, key, item) => {
     for (let i = 0; i < array.length; i++) {
-        if (array[i] === item) return i;
+        if ((array[i][key] === item) || (array[i] === item)) {
```

Comment:
What about this array?

```
[
  {},
  1,
  {
    pupa: {
      lupa: 1
    }
  },
  0,
  [1, 2, 3]
]
```

And if i gonna search such elements?

```
[1, 2, 3]
```
or
```
{}
```

Will I find something in your opinion?

Reply:
how was the key supposed to influence the search?
you do not need to search for something inside the element that you will find! in array!
U just need opportunity to find object in array